### PR TITLE
feat(Anomaly-Detection): add Half-Space Trees

### PR DESCRIPTION
**Pull request by justinuliu**
Add wrapper for Half-Space Trees for MOA.

### CODE REVIEW

**Comment by hmgomes** [comment]
Hi @justinuliu , excellent work getting this off the ground so quickly! 
I can see you haven't modified any of the high-level evaluation functions to incorporate anomaly detection, which is fine because we need to discuss how to do it. For now, I will just ask you to add a short notebook called `anomaly_detection.ipynb` demonstrating the functionality of the `HalfSpaceTree` and the `AUCEvaluator` in a test-then-train loop (you may use the first cell in `00_getting_started.ipynb` as a template). 

Cheers, 
Heitor

**Comment by justinuliu** [comment]
It seems that Half-Space Tree doesn't use seeds to initialize a random generator, so the AUC score differs with the same settings(including the same seed value).

**Comment by hmgomes** [comment]
> It seems that Half-Space Tree doesn't use seeds to initialize a random generator, so the AUC score differs with the same settings(including the same seed value). I think **this could ruin checks for future PRs**.

`HSTrees` is a subclass of `AbstractClassifier` thus it has a `classifierRandom` object which can have its random seed set via `randomSeedOption` in `AbstractClassifier`. Therefore, you can change the random seed of the learner like we do with `MOAClassifier` i.e.

```python
self.moa_learner.setRandomSeed(self.random_seed)
```

Different from other libraries, MOA does not randomly initialise random seeds if they are not set, it simply initialise them to a default value (See `AbstractClassifier`

```java
this.randomSeedOption = new IntOption("randomSeed", 'r', "Seed for random behaviour of the classifier.", 1);
```

File: src/capymoa/anomaly/half_space_trees.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+from capymoa.base import (
+    MOAClassifier,
+)
+
+from moa.classifiers.oneclass import HSTrees as _MOA_HSTrees
+
+
+class HalfSpaceTrees(MOAClassifier):
```
Comment (hmgomes) [inline src/capymoa/anomaly/half_space_trees.py:8]: Please add a MOAOutlier generic wrapper for anomaly detection methods and an abstract class Outlier. These will be similar, but different from the `Classifier` and `MOAClassifier`

**Comment by hmgomes** [comment]
I am going to go ahead and merge so that we have the first version of the anomaly detection API 👍

**Pull request merged**